streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -172,7 +174,14 @@ public String toString(final String indent) {
      * Flush all state stores owned by this task
      */
     void flushState() {
-        stateMgr.flush();
+        try {
+            stateMgr.flush();
+        } catch (final ProcessorStateException e) {
+            if (e.getCause() instanceof RetriableException) {
```

Comment:
Is this intentional? `ProducerFencedException` doesn't inherit from `RetriableException`, so unless I'm missing something, this will still kill the thread.  Same for below as well in `StandbyTask`.

Reply:
Oh, boy. You're right... Really off my game today.